src/components/App.jsx

```diff
@@ -71,18 +72,22 @@ class App extends React.Component {
   };
 
   getIssues = async event => {
-    event.preventDefault();
-    this.setState({ isEmpty: true, isFetching: true });
-    const finalUrl = this.createUrl();
-    await fetch(finalUrl)
-      .then(this.handleErrors)
-      .then(resJson => {
-        this.setState({ isEmpty: false, isFetching: false, results: resJson, url: finalUrl });
-      })
-      .catch(err => {
-        console.error('error:', err);
-        this.setState({ fetchError: true, isFetching: false });
-      });
+      event.preventDefault();
+
+    if (!this.state.isFetching && !this.state.buttonLock) {
```

Comment:
Do you mind de-structuring these? Just want to keep it consistent with the rest of the project.

Reply:
oops, hoped you wouldn't notice any formatting issues, was using vim, and I didnt feel like changing the tab rule, so I tried fixing anything manually.

Reply:
oh, you didnt notice it, misread it, but I caught a formatting issue anyway